applications/FluidDynamicsApplication/custom_processes/embedded_skin_visualization_process.cpp

```diff
@@ -620,23 +624,33 @@ void EmbeddedSkinVisualizationProcess::CreateVisualizationGeometries()
     for (int i_elem = 0; i_elem < n_elems; ++i_elem){
         ModelPart::ElementIterator it_elem = mrModelPart.ElementsBegin() + i_elem;
 
-        // Get element geometry
+        // Get element geometry and distances
         const Geometry<Node<3>>::Pointer p_geometry = it_elem->pGetGeometry();
         const unsigned int n_nodes = p_geometry->PointsNumber();
         const Vector nodal_distances = this->SetDistancesVector(it_elem);
+        const Vector edge_distances_extrapolated = this->SetEdgeDistancesExtrapolatedVector(it_elem);
 
-        // Check if the element is split
+        // Check if the element is split or Ausas incised
         const double zero_tol = 1.0e-10;
         const double nodal_distances_norm = norm_2(nodal_distances);
         if (nodal_distances_norm < zero_tol) {
             KRATOS_WARNING_FIRST_N("EmbeddedSkinVisualizationProcess", 10) << "Element: " << it_elem->Id() << " Distance vector norm: " << nodal_distances_norm << ". Please check the level set function." << std::endl;
         }
+        const bool is_ausas_incised = this->ElementIsAusasIncised(edge_distances_extrapolated);
         const bool is_split = this->ElementIsSplit(p_geometry, nodal_distances);
 
-        // If the element is split, create the new entities
-        if (is_split){
-            // Set the split utility and compute the splitting pattern
-            ModifiedShapeFunctions::Pointer p_modified_shape_functions = this->SetModifiedShapeFunctionsUtility(p_geometry, nodal_distances);
+        // If the element is split or Ausas incised, create the new entities
+        if (is_split) {
+            // Set the split utility
+            // NOTE: Check if is_ausas_incised needs to be done first, because ElementIsSplit() also returns 'true' for Ausas incised elements!
```

Comment:
I'm afraid of the future headaches caused by this... Just wondering if we should change the implementation of the function. Do you think it's feasible?

Reply:
Which part do you mean or what would you like to change about it?

Reply:
The point about the requirement of checking first the incised and then the split (not in the other way around). I think that for the moment we can do something similar as in the element (the split checks the incised internally).

Reply:
Actually, an incised element is not completely cut/ intersected, but it definitely is split.
The alternative would be to let ElementIsSplit() return wrong for incised elements and check ElementIsIncised() afterwards, just like yo suggested. Then we would have 150 lines of code double and we would call  ElementIsIncised() twice for incised elments (like in the element). 
Of course we could put these 150 lines of code into a separate method? Maybe that would aid the readability as well.

Reply:
OK. If we clearly document this in the code I'm fine.

Reply:
So, keep it like it is or check incised in ElementIsSplit() and separate method?

Reply:
Up to you. But my personal opinion would be to keep it as it is to avoid duplicating code.